e2e/specs/st_date_input.spec.js

```diff
@@ -294,4 +294,33 @@ describe("st.date_input", () => {
         "Date Input Changed: False"
     );
   });
+
+  it("renders the calendar component correctly", () => {
+    cy.get(".stDateInput").each((el, idx) => {
```

Comment:
@sfc-gh-jgarcia 
Perhaps time needs to pass for the changes to take effect. We can just change the theme to dark then process each calendar, and then change the theme to light and process the calendar. This allows the time to pass, and we don't worry too much about switching between light and dark mode regularly.

```javascript
// Get dark mode snapshot first. Taking light mode snapshot first
// for some reason ends up comparing dark with light
cy.changeTheme("Dark");

cy.get(".stDateInput").each((el, idx) => {
  if (idx === 5) {
    // Disabled one cannot be clicked
    return;
  }
  const testName = `date_input_calendar_${idx}`;
  cy.getIndexed(".stDateInput", idx).click();
  cy.get('[data-baseweb="calendar"]').matchImageSnapshot(
    `${testName}-dark`,
    {
      force: false,
    }
  );
});

// Revert back to light mode
cy.changeTheme("Light");
cy.get(".stDateInput").each((el, idx) => {
  if (idx === 5) {
    // Disabled one cannot be clicked
    return;
  }
  const testName = `date_input_calendar_${idx}`;
  cy.getIndexed(".stDateInput", idx).click();
  cy.get('[data-baseweb="calendar"]').matchImageSnapshot(
    testName,
    {
      force: false,
    }
  );
  cy.screenshot();
});
```